dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -502,6 +502,28 @@ object TypedDataset {
     new TypedDataset[A](reshaped.as[A](TypedExpressionEncoder[A]))
   }
 
+  def fromDF[A: TypedEncoder](df: DataFrame): TypedDataset[A] = {
```

Comment:
brilliant! I did stop for a minute looking at the duplication :) but I didn't think of this. Let me give it a shot